project5/3700kvstore

```diff
@@ -93,50 +144,56 @@ class Server:
         return len(self.replicas)//2 + 1
     
     def transition_to_state(self, state, msg=None, term=None):
-        if self.state != state:
-            logging.info('Term {} {} -> {}'.format(str(self.current_term),
-                str(self.state), str(state)))
-
-        if state == State.LEADER:
-            logging.info('Voted for by {}'.format(str(list(self.votes))))
-            # Re-initialize any necessary fields
-            self.state = State.LEADER
-            self.leader = self.id
-            self.prev_heartbeat = current_time()
-            for replica in self.replicas:
-                # For each server, index of next log entry to send to that server
-                # (initialized to leader last log index + 1)
-                self.next_idx[replica] = self.commit_idx + 1
-                # For each server, index of highest log entry known to be 
-                # replicated on server (initialized to 0, increases monotonically)
-                self.match_idx[replica] = 0
-
-            self.broadcast_heartbeat()
-            
-        elif state == State.FOLLOWER:
-            self.state = State.FOLLOWER
-            if msg:
-                self.leader = msg['leader']
-                self.current_term = msg['term']
-            self.votes = set()
-            self.voted_for = None
-
-        elif state == State.CANDIDATE:
-            self.state = State.CANDIDATE
-            self.leader = 'FFFF' # ID of the assumed leader
-            self.reset_timeouts()
-            if term is None:
-                self.current_term += 1
+        prev_state = self.state
+        prev_term = self.current_term
+        with self.accumulate_logs() as bypass_log:
+            if state == State.LEADER:
```

Comment:
Should we reset `self.votes()` and `self.voted_for` when converting to the leader?